packages/library/src/commonMain/kotlin/io/realm/RealmObject.kt

```diff
@@ -89,3 +90,31 @@ public fun RealmObject.isValid(): Boolean {
         true
     }
 }
+
+public fun <T : RealmObject> RealmObject.addChangeListener(callback: Callback<T?>): Cancellable {
+    checkNotificationsAvailable()
+    val realm = ((this as RealmObjectInternal).`$realm$Owner` as RealmReference).owner
+    @Suppress("UNCHECKED_CAST")
+    return realm.addObjectChangeListener(this as T, callback)
+}
+
+public fun <T : RealmObject> T.observe(): Flow<T?> {
+    checkNotificationsAvailable()
+    val internalObject = this as RealmObjectInternal
+    @Suppress("UNCHECKED_CAST")
+    return (internalObject.`$realm$Owner` as RealmReference).owner.observeObject(this as T)
+}
+
+private fun RealmObject.checkNotificationsAvailable() {
```

Comment:
Couldn't we split all the notification registration methods into its own interface and implement it only in the supporting classes and then just include a check for `owner is NotifiableRealm`. And then in all the calling methods we could just dispatch it by delegation. Probably for a separate PR that splits Realm into its interface and implementation.

Reply:
Yes, sounds like something we should think about, but will probably be overkill for this PR